### ISSUE TITLE
When open the source code of Flutter itself in VSCode, encounter an 'uri_does_not_exist' error.

### ISSUE THREAD

**Issue by shingo-nakanishi** [flutter/flutter#141529]
I am developing an app using Flutter.

I have an interest in Flutter's development itself (contributing).

I referred to https://github.com/flutter/flutter/wiki/Setting-up-the-Framework-development-environment and did a git clone.

I was able to execute up to [Example](https://github.com/flutter/flutter/wiki/Running-examples) introduced at https://github.com/flutter/flutter/wiki/Setting-up-the-Framework-development-environment.

Also, I have confirmed that the cloned Flutter version (Flutter 3.19.0-5.0.pre.42) works fine.

However, when I open this cloned folder in VSCode, I encounter an 'uri_does_not_exist' error.

For instance, this happens at the following places(The number of errors is 439K+ in VSCode.):

```
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
```

Since I can run the [Example](https://github.com/flutter/flutter/wiki/Running-examples), it seems that VSCode is not recognizing `package:` correctly.

Could you tell me how to solve this problem?

I've seen things like the following, but since they're not about Flutter itself, but answers for typical Flutter app developers, there were no answers for contributing to Flutter itself.

https://stackoverflow.com/questions/44909653/visual-studio-code-target-of-uri-doesnt-exist-packageflutter-material-dart

**Comment by huycozy** [comment]
Hi @shingo-nakanishi 
Are you seeing the error in [hello_world](https://github.com/flutter/flutter/tree/master/examples/hello_world) example? Did you run `flutter pub get` from the root dir of the project? Does this issue happen on other IDE like Android Studio?

It seems this issue is best suited to ask in Discord, please also refer to flutter's [community](https://flutter.dev/community) for resources.

**Comment by shingo-nakanishi** [comment]
@huycozy 

> Are you seeing the error in [hello_world](https://github.com/flutter/flutter/tree/master/examples/hello_world) example?
> 

I wasn't paying attention to whether there were any errors because it was working fine. I checked just now. It doesn't seem like there are any errors.

```
hello_world % ../../bin/flutter run
Connected devices:
macOS (desktop) • macos  • darwin-arm64   • macOS 13.5.1 22G90 darwin-arm64 (Rosetta)
Chrome (web)    • chrome • web-javascript • Google Chrome 120.0.6099.216

No wireless devices were found.

[1]: macOS (macos)
[2]: Chrome (chrome)
Please choose one (or "q" to quit): 2
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...              5.5s
This app is linked to the debug service: ws://127.0.0.1:56744/bpwdTU-9JSg=/ws
Debug service listening on ws://127.0.0.1:56744/bpwdTU-9JSg=/ws

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

A Dart VM Service on Chrome is available at: http://127.0.0.1:56744/bpwdTU-9JSg=
The Flutter DevTools debugger and profiler on Chrome is available at:
http://127.0.0.1:9100?uri=http://127.0.0.1:56744/bpwdTU-9JSg=
Application finished.
```

> Did you run `flutter pub get` from the root dir of the project?
> 

I'm not sure which folder you're referring to, but it seems that I can't run `flutter pub get` in the folder I cloned from git in the first place.

```
flutter % ./bin/flutter pub get
Expected to find project root in current working directory.
```

> Does this issue happen on other IDE like Android Studio?
> 

I am primarily (and still am) an iOS app development engineer, familiar with using Xcode, but I'm not accustomed to developing Flutter with Android Studio (though I have occasionally developed Android apps). I develop Flutter using VSCode. I've never looked into how to develop Flutter using Android Studio so it would take me some time to prepare.

**Comment by shingo-nakanishi** [comment]
@huycozy 

I'm beginning to understand the structure.

```
$ flutter/packages/flutter
```
```
$ ../../bin/flutter pub get
```

After doing the above, opening `/flutter/packages/flutter` in VSCode instead of `/flutter`

The `uri_does_not_exist` error has disappeared. Thank you.
